dark/views/tournament/add.py

```diff
@@ -16,22 +18,28 @@ class AddTournamentView(CreateView):
     def form_valid(self, form):
         creator = self.request.user
 
-        self.object = form.save(commit=False)
-        self.object.creator = creator
-        self.object.save()
+        try:
+            self.object = form.save(commit=False)
+            self.object.creator = creator
+            self.object.save()
 
-        tournament = Tournament.objects.get(pk=self.object.id)
+            tournament = Tournament.objects.get(pk=self.object.id)
 
-        tournament.participants.add(creator)
+            tournament.participants.add(creator)
 
-        number_of_teams = form.cleaned_data.get('number_of_teams')
+            number_of_teams = form.cleaned_data.get('number_of_teams')
+
+            for i in range(number_of_teams):
+                team_name = 'Team ' + str(i + 1)
+                team = Team.objects.create(tournament_id=self.object.id, name=team_name)
+                TeamRole.objects.create(team=team, name="Participant")
+
+            return redirect(self.get_success_url())
+        except IntegrityError as e:
+            messages.error(self.request, f'Failed to add tournament - make sure that field values are valid')
+            return super(AddTournamentView, self).form_invalid(form)
```

Comment:
Wiem że to inny typ błędu, ale też to tutaj wskaże:
jak podasz nie typ daty (np liczbę) w polu daty wywala NonType error
![image](https://user-images.githubusercontent.com/44002074/121253900-f166b380-c8a9-11eb-9e43-ba46993f6ae4.png)